operator/pkg/helmreconciler/wait.go

```diff
@@ -120,6 +124,13 @@ func waitForResources(objects object.K8sObjects, cs kubernetes.Interface, l *pro
 			if err != nil || newReplicaSet == nil {
 				return false, nil, nil, err
 			}
+			hpa, err = cs.AutoscalingV2().HorizontalPodAutoscalers(o.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
```

Comment:
Will we have a situation where deployment has been deployed, but hpa has not been applied yet? This will lead to the false positive issue as well.

Reply:
Although helm installation resources are [sequential](https://github.com/helm/helm/blob/2745909d3d753d68b7205c1f0a1153f50eb795b3/pkg/releaseutil/kind_sorter.go#L31), helm installation does not wait for the previous resource to be completely completed before applying the next one. It can be considered that the time spent on applying yaml can be ignored compared to the time on deployment ready.